### ISSUE TITLE
[BUG] All VMs created on one node in two-node cluster

### ISSUE THREAD

**Issue by simonflood** [harvester/harvester#1737]
**Describe the bug**
Yesterday I installed a two-node Harvester 1.0.0 cluster and now I'm loading it up with images and creating VMs. Having now created 8 VMs I'm surprised that they are all on one of the nodes and not split across both nodes. I've given each VM 1x CPU and both nodes are quad core with Hyper Threading enabled so 8 threads.

**To Reproduce**
1. Install two-node cluster
2. Create and launch 8 VMs
3. Note all 8 VMs are running on one node

**Expected behavior**
Whilst maybe not an equal split (4 and 4) across both nodes I did expect to see at least one running on the second node.

**Support bundle**
[supportbundle_afac18dc-92db-412a-922d-2b86a05e199d_2021-12-22T15-14-34Z.zip](https://github.com/harvester/harvester/files/7763460/supportbundle_afac18dc-92db-412a-922d-2b86a05e199d_2021-12-22T15-14-34Z.zip)

**Environment:**
 - Harvester ISO version: 1.0.0
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Baremetal with Intel NUC7i7DNxE with Hyper Threading enabled in BIOS

**Additional context**
I can't help wondering if it's related to the second unused node earlier being stuck in Maintenance mode? See #1731

**Comment by simonflood** [comment]
And a 9th VM also started on first NUC so that rules out a thought that maybe it was matching CPU cores/threads then switch but obviously not. Must be related to over-provisioning though better to load balance so not to have all eggs in the one basket.

**Comment by simonflood** [comment]
And a 10th and now finally Windows Server VM as 11th VM also on first NUC. Wonder how many I have to spin up before hitting a limit and either bringing second NUC into play?

**Comment by simonflood** [comment]
Having set the first NUC to maintenance mode and shut it down to upgrade RAM I still wasn't able to start VMs on the second NUC. The second NUC is the first one I installed in the cluster so perhaps something is set/cached?

At the moment I've got a 4-core NUC, now with 64GB RAM, that I can't start any VMs on which is a shame/waste. Thinking about reinstalling it (or both).

**Comment by guangbochen** [comment]
are u creating all of the VMs simultaneously via  the `Multiple Instance` option, if yes then there is a similar issue of https://github.com/harvester/harvester/issues/1824 and will be fixed via adding the soft-affinity rules.

**Comment by simonflood** [comment]
No, at that point I was creating VMs one at a time and they were all being created on the same (second) node.

**Comment by rrajendran17** [comment]
@simonflood Do you remember what network type was added to the VM when creating them.
There are other factors like NAD/Vlan VM Network attached to it which will be using a vlanconfig(physical nic) not spanning all nodes but some particular node.
In that case, the VMs will be spawned only on those nodes due to node affinity created.
